Fix #8504: ChartsJS scale and axes fixes

melloware opened the pull request.

melloware committed:
Fix #8504: ChartsJS scale and axes fixes

melloware commented:
I need to update the migration guide...

viiez commented on primefaces/src/main/java/org/primefaces/component/charts/ChartRenderer.java:
```diff
@@ -164,7 +164,15 @@ protected void encodeScales(FacesContext context, String chartName, Object scale
+                StringBuilder scaleAttrs = new StringBuilder(128);
+
+                if (cScales.getSuggestedMin() != null) {
+                    writeJsonAttribute(scaleAttrs, "suggestedMax", cScales.getSuggestedMin().toString());
```
```suggestion
                    writeJsonAttribute(scaleAttrs, "suggestedMin", cScales.getSuggestedMin().toString());
```

viiez commented on primefaces/src/main/java/org/primefaces/model/charts/axes/cartesian/CartesianScaleTitle.java:
```diff
@@ -217,7 +237,8 @@ public String encode() throws IOException {
 
             ChartUtils.writeDataValue(fsw, "display", this.display, false);
-            ChartUtils.writeDataValue(fsw, "labelString", this.labelString, true);
+            ChartUtils.writeDataValue(fsw, "title", this.title, true);
```
```suggestion
            ChartUtils.writeDataValue(fsw, "text", this.title, true);
```
https://www.chartjs.org/docs/latest/axes/labelling.html

viiez commented on primefaces/src/main/java/org/primefaces/component/charts/ChartRenderer.java:
```diff
@@ -184,7 +192,7 @@ protected void encodeScales(FacesContext context, String chartName, Object scale
             else if (scales instanceof RadialScales) {
                 writer.write("\"scale\":{");
```
I'm not sure if the default scaleId ('r') has to be encoded here as well like for x and y scales
https://www.chartjs.org/docs/latest/axes/#default-scales

viiez commented on primefaces/src/main/java/org/primefaces/component/charts/ChartRenderer.java:
```diff
@@ -164,7 +164,15 @@ protected void encodeScales(FacesContext context, String chartName, Object scale
+                if (cScales.getSuggestedMax() != null) {
+                    writeJsonAttribute(scaleAttrs, "suggestedMax", cScales.getSuggestedMax().toString());
+                }
+
```
suggestedMin and -Max are per axis options, therefore they have to be encoded for each axis individually, i.e. https://www.chartjs.org/docs/latest/axes/#axis-range-settings

viiez commented on primefaces/src/main/java/org/primefaces/model/charts/axes/AxesScale.java:
```diff
@@ -34,6 +34,8 @@
     private boolean display = true;
     private Number weight;
+    private Number suggestedMax;
+    private Number suggestedMin;
```
Please see my 2nd comment in ChartRenderer.java.

Since these options are 'per axis', there is no way to configure them for either x or y axis here.

viiez reviewed (commented):
I finished the review. Hope my comments are helpful. Feel free to contact me if you have any questions.

melloware commented:
OK all changes made. Their docs were misleading...

I think I have to make more changes too.  Min and Max were also moved

I have updated your reproducer working with 12.0.0-SNAPSHOT to show its all working here:
[pf-8504.zip](https://github.com/primefaces/primefaces/files/8170870/pf-8504.zip)

viiez reviewed (approved):
Reviewed and tested it. Works perfectly for me, thanks a lot!
